src/library.cpp

```diff
@@ -571,6 +571,24 @@ double lammps_get_natoms(void *handle)
   return natoms;
 }
 
+/* ----------------------------------------------------------------------
+   return the total number of bonds in the system
+   useful before call to lammps_get_bonds() so can pre-allocate vector
+------------------------------------------------------------------------- */
+
+int lammps_get_nbonds(void *handle)
+{
+  LAMMPS *lmp = (LAMMPS *) handle;
+
+  int nbonds = static_cast<int>(lmp->atom->nbonds);
+  if (nbonds > 9.0e15) return 0; // TODO:XXX why not -1?
+  if (!lmp->force->newton_bond) nbonds *= 2;
```

Comment:
This causes `extract_global` and `lammps_get_nbonds` to return two different values.

Reply:
Changing to `double` in the get_nbonds() method:
```
double lammps_get_nbonds(void *handle)
{
  LAMMPS *lmp = (LAMMPS *) handle;

  double nbonds = static_cast<double>(lmp->atom->nbonds);
  if (nbonds > 9.0e15) return 0; // TODO:XXX why not -1?
  return nbonds;
}
```

yields crazy numbers of nbonds in the example ('bond count:  1188987200') which is a bit weird? Shouldn't it work like this? Or is the variable nbonds in the atom class differently defined as compared to natoms?

Reply:
That's probably because you forgot to declare the return type. By default ctypes uses `int`, so you would truncate the double.

```
    self.lib.lammps_get_nbonds.argtypes = [c_void_p]
    self.lib.lammps_get_nbonds.restype = c_double
```

Reply:
There should be no "get_nbonds()" functions added to the C-library interface, since the same functionality is available already https://docs.lammps.org/Library_properties.html#extract-system-settings
Adding "get_xxx()" convenience functions for everything will quickly become a maintenance nightmare.
The long term plan should be to get rid of "get_atoms()" as well (although that will be difficult, since that is likely used by many applications of the library interface) and instead break down the "lammps_extract_global()" function to a few more topical subsets, including one that allows to query atom style items (like bonds, angles, dihedrals, ellipsoids, bodies, etc.).
For the python interface it may be acceptable when using some automated way to map "get_nXXX()" to "extract_global('XXX')", but that is for @rbberger to decide as the current caretaker of the Python module. I strongly recommend against making these individual functions.

Reply:
Thank you very much for the many comments. Thanks @rbberger for the hint with the declaration. Actually `ctypes` is a bit new for me and I am glad that I learned something new again. I agree with @akohlmey and I will remove the `get_nbonds()` function because it is not really needed for the `gather_bonds()` function either.

Reply:
I have removed `get_nbonds()` and used `get_thermo('bonds')` in the example to get the number of bonds that make this function obsolete.